fluss-server/src/main/java/com/alibaba/fluss/server/zk/ZooKeeperClient.java

```diff
@@ -397,6 +398,30 @@ public Map<String, Long> getPartitionNameAndIds(TablePath tablePath) throws Exce
         return partitions;
     }
 
+    /** Get the partition and the id for the partitions of a table in ZK by partition spec. */
+    public Map<String, Long> getPartitionNameAndIdsBySpec(
+            TablePath tablePath,
+            List<String> partitionKeys,
+            ResolvedPartitionSpec partitionSpecFromRequest)
+            throws Exception {
+        Map<String, Long> matchPartitions = new HashMap<>();
+        Set<String> partitions = getPartitions(tablePath);
+
+        for (String partitionName : partitions) {
+            ResolvedPartitionSpec resolvedPartitionSpecFromZK =
+                    ResolvedPartitionSpec.fromPartitionName(partitionKeys, partitionName);
```

Comment:
Change to static import: `ResolvedPartitionSpec.fromPartitionName(partitionKeys, partitionName)` -> `fromPartitionName(partitionKeys, partitionName)`